### DISCUSSION TITLE
How to run theme on AWS Amplify

### DISCUSSION THREAD

**Discussion by davidbeckonline** [McShelby/hugo-theme-relearn#421] [Q&A] [answered]
Hello,

I am running into an error when trying to deploy a re-learn theme based website in AWS amplify.
Below the error message I am seeing during the built phase.

~~~~
2022-12-19T10:55:55.792Z [WARNING]: Error: add site dependencies: load resources: loading templates: "/codebuild/output/src531285136/src/123/themes/hugo-theme-relearn/layouts/_default/_markup/render-codeblock-math.html:1:1": parse failed: template: _default/_markup/render-codeblock-math.html:1: unclosed action
2022-12-19T10:55:55.794Z [ERROR]: !!! Build failed
2022-12-19T10:55:55.794Z [ERROR]: !!! Non-Zero Exit Code detected
~~~~

Any idea what might be causing this issue?

File 
`.../layouts/_default/_markup/render-codeblock-math.html`
looks as follows:

~~~~
{{- partial "shortcodes/math.html" (dict
  "context" .Page
  "content" .Inner
  "align"  (index .Attributes "align")
) }}
~~~~

**Comment by McShelby** [comment]
I can not reproduce this.

Please retry to build the project in a local directory. If the error still occurs, please give version information about your Hugo executable, the theme's version or even better a link to your repo.

If it works in a local directory, it has something to do with your environment. Probably an issue with your used Hugo version.

As set in the theme's `config.toml` this theme requires at least Hugo 0.95.0

Quick googling: [Probably this](https://discourse.gohugo.io/t/error-unexpected-unclosed-action-in-command/36848/9?u=mcshelby)

**Comment by davidbeckonline** [comment]
You are right.
I can publish the website successfully in my AWS Cloud9 environment running hugo v0.108.0+extended.
The issue only comes up when trying to host the website via AWS Amplify.
Diving deeper into the logs, I am seeing the following messages:

`2022-12-20T08:35:53.530Z [WARNING]: ! Unable to write cache: {"code":"ERR_BAD_REQUEST","message":"Request failed with status code 404"})}`

`2022-12-20T08:35:53.583Z [WARNING]: !Failed to set up process.env.secrets`

`2022-12-20T08:36:57.554Z [INFO]: WARN 2022/12/20 08:36:57 Module "hugo-theme-relearn" is not compatible with this Hugo version; run "hugo mod graph" for more information.`

`2022-12-20T08:36:57.571Z [WARNING]: Error: add site dependencies: load resources: loading templates: "/codebuild/output/src090518717/src/[...]/themes/hugo-theme-relearn/layouts/_default/_markup/render-codeblock-math.html:1:1": parse failed: template: _default/_markup/render-codeblock-math.html:1: unclosed action
_default/_markup/render-codeblock-math.html:1: unclosed action
2022-12-20T08:36:57.573Z [ERROR]: !!! Build failed
2022-12-20T08:36:57.573Z [ERROR]: !!! Non-Zero Exit Code detected`

As a next step, I will reach out to AWS support in order to understand what HUGO version AWS Amplify is running.

**Comment by davidbeckonline** [comment, marked as answer]
I was able to resolve the issue by overwriting the default build settings of AWS Amplify.

A description on how to do this is given here:
https://gohugo.io/hosting-and-deployment/hosting-on-aws-amplify/

Some private blog posts indicate that Amplify is running on HUGO version 0.75.1 by default. I cannot confirm this information, but it would explain the error message I am seeing.
https://seanrmurphy.medium.com/deploying-and-managing-a-modern-hugo-site-on-aws-amplify-f771f6440b43
